src/components/accessories/patientOPD/patientOPDForm/PatientOPDForm.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { FunctionComponent, useCallback, useState } from 'react';
+import { useFormik } from "formik";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../../libraries/formDataHandling/functions";
+import DateField from '../../dateField/DateField';
+import { object } from 'yup';
+import { TProps } from './types';
+import ConfirmationDialog from '../../confirmationDialog/ConfirmationDialog';
+import TextButton from '../../textButton/TextButton';
+import SmallButton from '../../smallButton/SmallButton';
+import warningIcon from "../../../../assets/warning-icon.png";
+import TextField from '../../textField/TextField';
+import has from 'lodash.has';
+import get from 'lodash.get';
+import "./styles.scss"
+
+const PatientOPDForm: FunctionComponent<TProps> = ({
+  fields,
+  onSubmit,
+  submitButtonLabel,
+  resetButtonLabel,
+  isLoading,
+}) => {
+
+  const validationSchema = object({
+    // TODO  
+  });
+
+  const initialValues = getFromFields(fields, "value");
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      const formattedValues = formatAllFieldValues(fields, values);
+      onSubmit(formattedValues);
+    },
+  });
+
+  const { setFieldValue } = formik;
+
+  const dateFieldHandleOnChange = useCallback(
+    (value) => {
+      setFieldValue("opdDate", value);
```

Comment:
Here we could apply a curried function, as suggested at [this comment](https://github.com/informatici/openhospital-ui/pull/81#discussion_r567220664)